.yarnrc.yml

```diff
@@ -9,8 +9,14 @@ npmRegistryServer: "https://registry.yarnpkg.com"
 plugins:
   - path: .yarn/plugins/@yarnpkg/plugin-typescript.cjs
     spec: "@yarnpkg/plugin-typescript"
+  - path: .yarn/plugins/@yarnpkg/plugin-constraints.cjs
+    spec: "@yarnpkg/plugin-constraints"
+  - path: .yarn/plugins/@yarnpkg/plugin-workspace-tools.cjs
+    spec: "@yarnpkg/plugin-workspace-tools"
+  - path: .yarn/plugins/@yarnpkg/plugin-version.cjs
+    spec: "@yarnpkg/plugin-version"
 
 unsafeHttpWhitelist:
   - localhost
 
-yarnPath: .yarn/releases/yarn-sources.cjs
+yarnPath: .yarn/releases/yarn-2.4.2.cjs
```

Comment:
I'd suggest going for `3.0.0-rc.6` (or any of the RC versions) and reimport the plugins to get the various fixes to the constraints plugin and general performance improvements